vm/runtime/src/lib.rs

```diff
@@ -150,11 +153,42 @@ pub trait Syscalls {
     /// Computes an unsealed sector CID (CommD) from its constituent piece CIDs (CommPs) and sizes.
     fn compute_unsealed_sector_cid(
         &self,
-        _reg: RegisteredProof,
-        _pieces: &[PieceInfo],
+        proof_type: RegisteredProof,
+        pieces: &[PieceInfo],
     ) -> Result<Cid, ActorError> {
-        // TODO
-        todo!()
+        let mut sum = 0u64;
+        for p in pieces {
+            sum += p.size.0;
+        }
+
+        let ssize = proof_type.sector_size() as u64;
+
+        let mut fcp_pieces: Vec<filecoin_proofs_api::PieceInfo> = pieces
+            .iter()
+            .map(filecoin_proofs_api::PieceInfo::try_from)
+            .collect::<Result<_, &'static str>>()
```

Comment:
```suggestion
                        .collect::<Result<Vec<filecoin_proofs_api::PieceInfo>, &'static str>>()
```
Then you can remove the type annotation on fcp_pieces. This is a little more understandable to me, BUT I could be in the minority... Up to you :)

Reply:
Idk I thought this way is more readible because it's more clear what comes out of it

Reply:
That sounds pretty logical!